consensus/state.go

```diff
@@ -686,6 +699,12 @@ func (cs *State) newStep() {
 
 	cs.nSteps++
 
+	cs.eventCollector.WritePoint(consensusTable, map[string]interface{}{
+		"height": rs.Height,
+		"round":  rs.Round,
+		"step":   rs.Step,
+	})
```

Comment:
would we be able to collect arbitary trace data that way? Like, if I wanted a summary of each message sent and recieved over the wire, would we be able to put that in a gauge or counter?

I'm not exactly sure of how we will use this yet. Initially, I assumed that we would create proprietary one-off branches that push very specific trace data to the db for us to examine some scenario or component. However I could see have some default set of data collection that could be turned on simply by configuring a url and token to also be useful.

Reply:
I've since removed the above code, so by default we're not actually sending any data to the db even if we have it configured. However we can add things in the future!

https://github.com/celestiaorg/celestia-core/pull/970/commits/6758a3e152c7e3892cd244d365fae76851dd5091

Reply:
> Thus main should also have the same data tracked.

> I think that often these will be the same data points, thus by just capturing all metrics by default, we save having to rewrite this every time we want to analyse something.

I agree, but wasn't sure how much data we want to collect atm and felt like that could be pushed to a different PR. 

> However we can add things or refactor in the future!

added #978 

For example, I think collecting the consensus info makes sense here, but what about a trace of all of the messages sent over the wire? That seems both very useful in some cases, but also very heavy, so idk.

Reply:
>  Instead of duplicating ourselves, can we implement the metrics.Gauge and metrics.Counter interfaces using this eventCollector and then in the metrics.go file of each reactor, include a InfluxDBMetrics function alongside the PrometheusMetrics function.

also sorry I think I mistunderstood this at first to say that we should refactor the eventCollector to use guages and counters for collection. being able to keep everything in a single db is a good idea :sweat_smile:

Reply:
> That seems both very useful in some cases, but also very heavy, so idk.

We could look at filtering metrics by reactor (since they're already divided that way) similar to the filtered logger Tendermint had in the past